### PR TITLE
Event filtering using the backend

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
This PR introduces filtering logic for events in the backend, as well as the Console making use of the new feature. Client-side filtering is hence retired.

Referencing #2231 

#### Changes
- Add server-side filtering via the `names` prop
- Removing client-side filtering from the Console
- Updating structure of event filters, and associated handling logic in the Console
  - Using specific per-entity filters to avoid server overhead
  - Using a more sustainable

#### Testing

Manual testing.

##### Regressions

Possible but hard to check right now. We should check this closely once on staging.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/events/grpc/grpc.go [hunk 1]
```diff
@@ -41,16 +46,74 @@ func NewEventsServer(ctx context.Context, pubsub events.PubSub) *EventsServer {
...
+	definedNames map[string]struct{}
+}
+
+var (
+	errInvalidRegexp    = errors.DefineInvalidArgument("invalid_regexp", "invalid regexp")
+	errNoMatchingEvents = errors.DefineInvalidArgument("no_matching_events", "no matching events for regexp `{regexp}`")
+	errUnknownEventName = errors.DefineInvalidArgument("unknown_event_name", "unknown event `{name}`")
+)
+
+func (srv *EventsServer) processNames(names ...string) ([]string, error) {
```
Comment (adriansmares) [inline pkg/events/grpc/grpc.go:73]: This is clever.

**Review by adriansmares — APPROVED** [review]

File: pkg/webui/console/lib/events/definitions.js [hunk 2]
```diff
@@ -57,15 +57,15 @@ export const eventMessages = defineMessages({
...
+    'The event stream will now be filtered, meaning that some event types may be suppressed. See the details below for more information about the currently enabled filter.',
 
-  'synthetic.status.verbose_disabled:type': 'Verbose stream deactivated',
-  'synthetic.status.verbose_disabled:preview': 'The event stream has been set to non-verbose',
-  'synthetic.status.verbose_disabled:details':
-    'The event stream verbosity has been set to normal. The stream will now start to discard certain advanced event types. This will help reducing noise in the event stream. You can see the currently applied filter in the raw data of this event.',
+  'synthetic.status.filter_disabled:type': 'Filter disabled',
+  'synthetic.status.filter_disabled:preview': 'The previously set event filter has been disabled',
+  'synthetic.status.filter_disabled:details':
+    'The event stream is will not be filtered any more. This means that you will see all events that come out of the event stream for this entity.',
```
Comment (bafonins) [inline pkg/webui/console/lib/events/definitions.js]:
```suggestion
    'The event stream will not be filtered any more. This means that you will see all events that come out of the event stream for this entity.',
```

**Review by bafonins — APPROVED** [review]

**Review by htdvisser — COMMENTED** [review]
I can't really review the code-owned files here, since I wrote those changes myself....

**Comment by kschiffer** [comment]
@johanstokking Can you review in this case since this PR is co-authored by @htdvisser. I cannot merge otherwise.

**Pull request merged**